middleware.ts

```diff
@@ -0,0 +1,57 @@
+import type { NextRequest } from "next/server";
+import { geolocation, ipAddress } from "@vercel/edge";
+import { NextResponse } from "next/server";
+
+const BLOCKED_COUNTRIES = [
+  "CU", // Cuba
+  "IR", // Iran
+  "KP", // North Korea
+  "SY", // Syria
+  "RU", // Russia
+  "unknown", // Block if we cannot identify country
+];
+
+const BLOCKED_REGIONS = [
+  { country: "UA", region: "14" }, // Ukraine Donetsk
+  { country: "UA", region: "43" }, // Ukraine Crimea
+  { country: "UA", region: "09" }, // Ukraine Luhansk
+  { country: "UA", region: "unknown" }, // Ukraine if we cannot identify region
```

Comment:
@musnit Blocking Ukraine if we cannot identify the region. Basically I am not sure how good the region detection is. For example for my IP I see:

```
  city: undefined,
  countryRegion: undefined,
```

So playing it safe. We have server side logging to see if we hit this case.